### DISCUSSION TITLE
Reading large text file

### DISCUSSION THREAD

**Discussion by retog** [denoland/deno#17024] [Q&A]
Reading a larget file with `Deno.readTextFile` I get:

```
TypeError: Cannot allocate String: buffer exceeds maximum length.
at async Object.readTextFile (deno:runtime/js/40_read_file.js:56:20)
```

Can this maximum length be increased?

**Comment by bartlomieju** [comment]
What is the size of the file?

**Reply by retog** [reply]
1.4 GB

**Reply by bartlomieju** [reply]
@littledivy please take a look, IIRC the limit for string size is 4Gb, so we should be hitting this error

**Comment by retog** [comment]
Hey @bartlomieju, @littledivy, any new idea or suggestion on how to deal with large files? Thanks for your help!

**Reply by littledivy** [reply]
@retog Ideally large files should be streamed in multiple chunks into memory.

```ts
const file = await Deno.open('tmp.txt');
const reader = file.readable.getReader();

while(true) {
  // Process each chunk
  const { value, done } = reader.read();
  /// ...
  if (done) break;
}
```
